houston/generator/base.py

```diff
@@ -199,12 +204,36 @@ def record_outcome(self, mission, outcome):
         """
         self.__history.append(mission)
         self.__outcomes[mission] = outcome
+        if coverage:
+            self.__coverage[mission] = coverage
 
         if outcome.failed:
             self.__failures.add(mission)
 
 
     def generate(self, seed, resource_limits):
+        """
```

Comment:
Since this is a generator, maybe it could implement `__iter__` and `__next__`, and use `__next__` to lazily construct a stream of missions?